tools/NuGetPackager/PackCommand.cs

```diff
@@ -0,0 +1,146 @@
+            {
+                throw new GracefulException("Build failed");
+            }
+
+            Directory.CreateDirectory(outputDir);
+
+            var version = project.ProjectFile.Version.ToNormalizedString();
+            await Nuget("pack",
+                nuspec,
+                "-Verbosity", "detailed",
+                "-OutputDirectory", outputDir,
+                "-Version", version,
+                "-Properties", props);
+        }
+
+        private IEnumerable<Tuple<string, string>> GetDependencies(ProjectContext context)
+        {
+            // copied from https://github.com/dotnet/cli/blob/rel/1.0.0/src/dotnet/commands/dotnet-pack/PackageGenerator.cs
+            var project = context.RootProject;
+
+            foreach (var dependency in project.Dependencies)
+            {
+                if (dependency.Type.Equals(LibraryDependencyType.Build))
+                {
+                    continue;
+                }
+
+                // TODO: Efficiency
+                var dependencyDescription = context.LibraryManager.GetLibraries().First(l => l.RequestedRanges.Contains(dependency));
+
+                // REVIEW: Can we get this far with unresolved dependencies
+                if (dependencyDescription == null || !dependencyDescription.Resolved)
+                {
+                    continue;
+                }
+
+                if (dependencyDescription.Identity.Type == LibraryType.Project &&
+                    ((ProjectDescription)dependencyDescription).Project.EmbedInteropTypes)
+                {
+                    continue;
+                }
+
+
+                VersionRange dependencyVersion = null;
+
+                if (dependency.VersionRange == null ||
+                    dependency.VersionRange.IsFloating)
+                {
+                    dependencyVersion = new VersionRange(dependencyDescription.Identity.Version);
+                }
+                else
+                {
+                    dependencyVersion = dependency.VersionRange;
+                }
+
+                Reporter.Verbose.WriteLine($"Adding dependency {dependency.Name.Yellow()} {VersionUtility.RenderVersion(dependencyVersion).Yellow()}");
+
+                yield return new Tuple<string, string>(dependency.Name, dependencyVersion.MinVersion.ToString());
+            }
+        }
+
+        private static string GetLockFileVersion(ProjectContext project, string name) =>
+            project
+                .LockFile
+                .PackageLibraries
+                .First(l => l.Name.Equals(name))
+                .Version
+                .ToNormalizedString();
+
+        private async Task Nuget(params string[] args)
+        {
+            var pInfo = new ProcessStartInfo
+            {
+                Arguments = ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart(args),
+                FileName = await GetNugetExePath()
```

Comment:
Yup. Running on xplat would require running it via mono (which we don't require and is known to be problematic).